### ISSUE TITLE
Generalized Request regresion 

### ISSUE THREAD

**Issue by dalcinl** [open-mpi/ompi#8391]
Thank you for taking the time to submit an issue!

### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)

4.1.0

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

Distribution tarball

`./configure --prefix=... --enable-debug --enable-mem-debug && make -j 32 && make install`

### Please describe the system on which you are running

* Operating system/version: Linux, Fedora 33
* Computer hardware: x86_64
* Network type: single node

-----------------------------

## Details of the problem

The following trivial mpi4py test for generalized requests fails:

```python
from mpi4py import MPI
greq = MPI.Grequest.Start(None, None, None)
assert greq.Test() == False
```
While mpi4py does a bit of special treatment if you pass `None` for the query/free/cancel callbacks, these details are irrelevant for this test. As the generalized request has not been marked as completed yet, no callbacks should be invoked and then `MPI_Test()` should return False. This looks like a regression in v4.1.0 (the test code runs OK under  v4.0.5)

**Comment by bosilca** [comment]
I think issue has been already addressed in #8340.

**Comment by dalcinl** [comment]
Yes, looks like this one is a duplicate, sorry for the noise.
